src/catalog-debug/src/main.rs

```diff
@@ -0,0 +1,519 @@
+    /// then passed to the `edit` or `delete` commands.
+    Dump {
+        /// Write output to specified path. Default stdout.
+        target: Option<PathBuf>,
+    },
+    /// Edits a single item in a collection in the stash.
+    Edit {
+        /// The name of the stash collection to edit.
+        collection: String,
+        /// The JSON-encoded key that identifies the item to edit.
+        key: serde_json::Value,
+        /// The new JSON-encoded value for the item.
+        value: serde_json::Value,
+    },
+    /// Deletes a single item in a collection in the stash
+    Delete {
+        /// The name of the stash collection to edit.
+        collection: String,
+        /// The JSON-encoded key that identifies the item to delete.
+        key: serde_json::Value,
+    },
+    /// Checks if the specified stash could be upgraded from its state to the
+    /// adapter catalog at the version of this binary. Prints a success message
+    /// or error message. Exits with 0 if the upgrade would succeed, otherwise
+    /// non-zero. Can be used on a running environmentd. Operates without
+    /// interfering with it or committing any data to that stash.
+    UpgradeCheck {
+        /// Map of cluster name to resource specification. Check the README for latest values.
+        cluster_replica_sizes: Option<String>,
+    },
+}
+
+#[tokio::main]
+async fn main() {
+    let args = cli::parse_args(CliConfig {
+        env_prefix: Some("MZ_CATALOG_DEBUG_"),
+        enable_version_flag: true,
+    });
+    if let Err(err) = run(args).await {
+        eprintln!("catalog: fatal: {}", err.display_with_causes());
+        process::exit(1);
+    }
+}
+
+async fn run(args: Args) -> Result<(), anyhow::Error> {
+    let metrics_registry = MetricsRegistry::new();
+    let openable_state: Box<dyn OpenableDurableCatalogState> = match args.store {
+        CatalogKind::Stash => {
+            let postgres_url = args.postgres_url.expect("required for stash");
+            let tls =
+                mz_tls_util::make_tls(&tokio_postgres::config::Config::from_str(&postgres_url)?)?;
+            let factory = StashFactory::new(&metrics_registry);
+            let stash_config = StashConfig {
+                stash_factory: factory,
+                stash_url: postgres_url,
+                schema: None,
+                tls,
+            };
+            Box::new(stash_backed_catalog_state(stash_config))
+        }
+        CatalogKind::Persist => {
+            let persist_config =
+                PersistConfig::new(&mz_environmentd::BUILD_INFO, SYSTEM_TIME.clone());
+            let persist_pubsub_server =
+                PersistGrpcPubSubServer::new(&persist_config, &metrics_registry);
+            let persist_pubsub_client = persist_pubsub_server.new_same_process_connection();
+
+            let persist_clients =
+                PersistClientCache::new(persist_config, &metrics_registry, |_, metrics| {
```

Comment:
If this is just a cli tool, then I'd skip the pubsub stuff, it wouldn't get you anything

Reply:
How would I create a `PersistClientCache` without the pubsub stuff?

Reply:
Something like this?

```
            let persist_config =
                PersistConfig::new(&mz_environmentd::BUILD_INFO, SYSTEM_TIME.clone());
            let persist_clients =
                PersistClientCache::new(persist_config, &metrics_registry, |_, metrics| {
                    PubSubClientConnection::noop()
                });
```

Reply:
Yeah that looks right